domains/transfer-agreements/TransferAgreements.API/API/Controllers/TransferAgreementsController.cs

```diff
@@ -12,24 +13,27 @@ namespace API.Controllers;
 [ApiController]
 public class TransferAgreementsController : ControllerBase
 {
-    private readonly ITransferAgreementService transferAgreementService;
+    private readonly ITransferAgreementRepository transferAgreementRepository;
 
-    public TransferAgreementsController(ITransferAgreementService transferAgreementService) => this.transferAgreementService = transferAgreementService;
+    public TransferAgreementsController(ITransferAgreementRepository transferAgreementRepository) => this.transferAgreementRepository = transferAgreementRepository;
 
     [HttpPost("api/transfer-agreements")]
     public async Task<ActionResult> Create([FromBody] CreateTransferAgreement request)
     {
+        var actor = User.FindActorClaim();
+        var subject = User.FindSubjectClaim();
+
         var transferAgreement = new TransferAgreement
         {
-            SenderId = Guid.Parse(User.FindFirstValue("sub")),
-            ActorId = User.FindFirstValue("atr"),
-            StartDate = request.StartDate.UtcDateTime,
-            EndDate = request.EndDate.UtcDateTime,
+            StartDate = DateTimeOffset.FromUnixTimeSeconds(request.StartDate),
+            EndDate = DateTimeOffset.FromUnixTimeSeconds(request.EndDate),
+            ActorId = actor,
+            SenderId = Guid.Parse(subject),
             ReceiverTin = request.ReceiverTin
         };
 
-        var result = await transferAgreementService.CreateTransferAgreement(transferAgreement);
+        var result = await transferAgreementRepository.CreateTransferAgreement(transferAgreement);
 
-        return Ok(result);
+        return Created($"api/transfer-agreements/{result.Id}", result);
```

Comment:
You are hardcoding the route, which should be fixed now or at a later point. If you had the GET endpoint, then you could use `CreatedAtRoute` or `CreatedAtAction`.

Reply:
Noted. It will be fixed, while implementing the GET endpoint in #1465